src/GraphQl/Type/SchemaBuilder.php

```diff
@@ -516,8 +516,10 @@ private function getResourceObjectTypeFields(?string $resourceClass, ResourceMet
             return $fields;
         }
 
-        if (!$input || 'create' !== $mutationName) {
+        if (!$input || \in_array($mutationName, ['update', 'delete'], true)) {
             $fields['id'] = $idField;
+        } elseif ('create' !== $mutationName) {
+            $fields['id'] = ['type' => GraphQLType::id()];
```

Comment:
I don't understand the aim of this `elseif`.

Reply:
The idea is to still generate the `id` field for all mutations except `create`, but making it optional. This will allow custom mutations to mutate the specified resource, if provided - but also allows custom mutations that do not need an existing resource.

Reply:
OK I see. Could you make it an `$optionalIdField` variable to be clearer?

Reply:
Makes sense. I will make the change together with the test later today 🙂